Foundation/CPCache.j

```diff
@@ -0,0 +1,350 @@
+/*
+ * CPCache.j
+ * Foundation
+ *
+ * Created by William Mura.
+ * Copyright 2015, William Mura.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+ */
+
+
+@import "CPObject.j"
+
+/*
+ * Class _CPCacheItem
+ * Represent an item of CPCache
+ * This class allow to associate a cost and a position to an object
+ *
+ * Attributes:
+ * - object: the stored object
+ * - cost: represent the cost (of memory) of the object
+ * - position: represent the insertion order to determine the oldest object
+ */
+@implementation _CPCacheItem : CPObject
+{
+    CPObject    _object      @accessors(property=object);
+    int         _cost        @accessors(property=cost);
+    int         _position    @accessors(property=position);
+}
+
++ (id)cacheItemWithObject:(CPObject)anObject cost:(int)aCost position:(int)aPosition
+{
+    var cacheItem = [[super alloc] init];
+
+    if (cacheItem)
+    {
+        cacheItem._object = anObject;
+        cacheItem._cost = aCost;
+        cacheItem._position = aPosition;
+    }
+
+    return cacheItem;
+}
+
+@end
+
+
+/*
+ * Delegate CPCacheDelegate
+ *
+ * - cache:willEvictObject: is called when a object is going to be removed
+ * When the total cost or the count exceeds the total cost limit or the count limit
+ * And also when removeObjectForKey or removeAllObjects are called
+ */
+@protocol CPCacheDelegate <CPObject>
+
+@optional
+- (void)cache:(CPCache)cache willEvictObject:(id)obj;
+
+@end
+
+
+var CPCacheDelegate_cache_WillEvictObject = 1 << 1;
```

Comment:
style:

Should be `var  CPCacheDelegate_cache_willEvictObject_` (replace all `:` with `_`)